packages/navie/src/explain.ts

```diff
@@ -188,12 +197,28 @@ export default function explain(
     memoryService
   );
 
-  return {
-    on: (event: 'event', listener: (event: InteractionEvent) => void) => {
-      interactionHistory.on(event, listener);
-    },
-    execute() {
-      return codeExplainerService.execute(clientRequest, options, chatHistory);
-    },
-  };
+  class ExplainImplementation extends EventEmitter implements IExplain {
+    constructor() {
+      super();
+
+      interactionHistory.on('event', (event: InteractionEvent) => {
+        this.emit('event', event);
+
+        if (event instanceof AgentSelectionEvent) {
+          this.emit('agent', event.agent);
+        }
+
+        if (event instanceof ClassificationEvent) {
+          this.emit('classification', event.classification);
+        }
+      });
+    }
+
+    // eslint-disable-next-line class-methods-use-this
+    async *execute(): AsyncIterable<string> {
+      yield* codeExplainerService.execute(clientRequest, options, chatHistory);
+    }
```

Comment:
Similar to returning a `Promise`, I don't believe the `async`/`yield` is necessary. We're not consuming the iterator, so it can be returned directly.

```suggestion
    execute(): AsyncIterable<string> {
      return codeExplainerService.execute(clientRequest, options, chatHistory);
    }
```

OTOH, this is what it looked like previously so maybe there's something I'm missing.

Reply:
Yeah. Weirdly, it won't compile without the async.